boa/interpret.py

```diff
@@ -26,39 +28,30 @@
 _disk_cache = None
 
 
-class BoaImporter(importlib.abc.MetaPathFinder):
-    def __init__(self):
-        self._path_lookup = {}
-
-    # TODO: replace this with more modern `find_spec()`
-    def find_module(self, fullname, package_path, target=None):
+class BoaImporter(MetaPathFinder):
+    def find_spec(self, fullname, path, target=None):
         path = Path(fullname.replace(".", "/")).with_suffix(".vy")
 
         for prefix in sys.path:
             to_try = Path(prefix) / path
 
             if to_try.exists():
-                self._path_lookup[fullname] = to_try
-                return self
-
-        return None
+                loader = BoaLoader(fullname, str(to_try))
+                return spec_from_loader(fullname, loader)
 
-    # TODO: replace with more modern `exec_module()` and `create_module()`
-    def load_module(self, fullname):
-        if fullname in sys.modules:
-            return sys.modules[fullname]
 
-        # import system should guarantee this, but be paranoid
-        if fullname not in self._path_lookup:
-            raise ImportError(f"invariant violated: no lookup for {fullname}")
+class BoaLoader(SourceFileLoader):
+    def get_code(self, fullname):
+        return ""
```

Comment:
per https://docs.python.org/3/library/importlib.html#importlib.abc.InspectLoader.get_code, this should return None

Reply:
![image](https://github.com/vyperlang/titanoboa/assets/2369243/41794053-5255-46c9-9fd6-b12a07d9a7da)

Reply:
nvm, discussed offline -- apparently returning `None` triggers an `ImportError`.

Reply:
let's add a comment though
```suggestion
        # importlib docs say to return None, but that triggers an `ImportError`
        return ""
```